reference/test/solver/pipe_cg_kernels.cpp

```diff
@@ -0,0 +1,694 @@
+// SPDX-FileCopyrightText: 2025 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "core/solver/pipe_cg_kernels.hpp"
+
+#include <gtest/gtest.h>
+
+#include <ginkgo/core/base/exception.hpp>
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/matrix/dense.hpp>
+#include <ginkgo/core/solver/pipe_cg.hpp>
+#include <ginkgo/core/stop/combined.hpp>
+#include <ginkgo/core/stop/iteration.hpp>
+#include <ginkgo/core/stop/residual_norm.hpp>
+#include <ginkgo/core/stop/time.hpp>
+
+#include "core/test/utils.hpp"
+
+
+namespace {
+
+
+template <typename T>
+class PipeCg : public ::testing::Test {
+protected:
+    using value_type = T;
+    using Mtx = gko::matrix::Dense<value_type>;
+    using Solver = gko::solver::PipeCg<value_type>;
+    PipeCg()
+        : exec(gko::ReferenceExecutor::create()),
+          mtx(gko::initialize<Mtx>(
+              {{2, -1.0, 0.0}, {-1.0, 2, -1.0}, {0.0, -1.0, 2}}, exec)),
+          stopped{},
+          non_stopped{},
+          pipe_cg_factory(
+              Solver::build()
+                  .with_criteria(
+                      gko::stop::Iteration::build().with_max_iters(400u),
+                      gko::stop::Time::build().with_time_limit(
+                          std::chrono::seconds(6)),
+                      gko::stop::ResidualNorm<value_type>::build()
+                          .with_reduction_factor(r<value_type>::value))
+                  .on(exec)),
+          mtx_big(gko::initialize<Mtx>(
```

Comment:
the actual initiation follows the member declaration.
here you need to make it to follow the same order of member.